src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -88,14 +88,20 @@ Json::Value Profile::GenerateStub() const
         stub[JsonKey(GuidKey)] = winrt::to_string(Utils::GuidToString(*_Guid));
     }
 
-    stub[JsonKey(NameKey)] = winrt::to_string(_Name);
+    if (_Name)
+    {
+        stub[JsonKey(NameKey)] = winrt::to_string(*_Name);
+    }
 
-    if (!_Source.empty())
+    if (_Source && !_Source.value().empty())
     {
-        stub[JsonKey(SourceKey)] = winrt::to_string(_Source);
+        stub[JsonKey(SourceKey)] = winrt::to_string(*_Source);
     }
 
-    stub[JsonKey(HiddenKey)] = _Hidden;
+    if (_Hidden)
+    {
+        stub[JsonKey(HiddenKey)] = *_Hidden;
+    }

```

Comment:
stub is intended to _always contain these values_ -- make sure you always set them, even if they were inherited.